### ISSUE TITLE
Specific search term causes mpsyt to crash

### ISSUE THREAD

**Issue by KetchupBomb** [mps-youtube/mps-youtube#666]
When searching for `marshmellow`:

```
bradking-mbp:~ bradking$ mpsyt //marshmellow
Searching playlists for marshmellow
Traceback (most recent call last):
  File "/usr/local/bin/mpsyt", line 11, in <module>
    load_entry_point('mps-youtube==0.2.7.1', 'console_scripts', 'mpsyt')()
  File "/usr/local/Cellar/mps-youtube/0.2.7.1_4/libexec/lib/python3.6/site-packages/mps_youtube/main.py", line 141, in main
    if matchfunction(i.function, i.regex, userinput):
  File "/usr/local/Cellar/mps-youtube/0.2.7.1_4/libexec/lib/python3.6/site-packages/mps_youtube/main.py", line 64, in matchfunction
    func(*matches)
  File "/usr/local/Cellar/mps-youtube/0.2.7.1_4/libexec/lib/python3.6/site-packages/mps_youtube/commands/search.py", line 268, in pl_search
    qs['id'] = ','.join(id_list)
TypeError: sequence item 9: expected str instance, NoneType found
```

When searching for `marshmallow`:

```
bradking-mbp:~ bradking$ mpsyt //marshmallow
Item  Playlist                                                                                                                                                                                                                                                                                                                                 Author       Updated   Count
1     Marshmallow                                                                                                                                                                                                                                                                                                                              Anthony Onno 07/07/16    14
2     Marshmallow Mix                                                                                                                                                                                                                                                                                                                          Ali M        01/02/17    50
3     marshmallow mix                                                                                                                                                                                                                                                                                                                          Onur Tezer   10/26/16     9
...
```

---

[When searching YouTube via browser for `marshmellow`](https://www.youtube.com/results?search_query=marshmellow&page=&utm_source=opensearch), there is a spelling suggestion. Searching for `marshmallow` has no spelling suggestion. Potentially that is the culprit?

---

```
bradking-mbp:~ bradking$ mpsyt --version
mpsyt version      : 0.2.7.1
   notes           : released 6 July 2016
pafy version       : 0.5.3.1 (youtube-dl backend)
youtube-dl version : 2017.05.18.1
Python version     : 3.6.1 (default, Apr  4 2017, 09:40:21)
[GCC 4.2.1 Compatible Apple LLVM 8.1.0 (clang-802.0.38)]
Processor          : i386
Machine type       : x86_64
Architecture       : 64bit,
Platform           : Darwin-16.7.0-x86_64-i386-64bit
sys.stdout.enc     : UTF-8
default enc        : utf-8
Config dir         : /Users/bradking/.config/mps-youtube
env:TERM           : screen
env:SHELL          : /bin/bash
env:LANG           : en_US.UTF-8
```

**Comment by ritiek** [comment]
Duplicate of #663. This should be fixed on the `develop` branch (#664).

Apparently, it was happening due to channel results being mixed with video and playlist results.

**Comment by drzacek** [comment]
$ mpsyt //green day
Searching playlists for green day                                                                       
Traceback (most recent call last):
  File "/usr/local/bin/mpsyt", line 11, in <module>
    sys.exit(main.main())
  File "/usr/local/lib/python3.5/dist-packages/mps_youtube/main.py", line 141, in main
    if matchfunction(i.function, i.regex, userinput):
  File "/usr/local/lib/python3.5/dist-packages/mps_youtube/main.py", line 64, in matchfunction
    func(*matches)
  File "/usr/local/lib/python3.5/dist-packages/mps_youtube/commands/search.py", line 268, in pl_search
    qs['id'] = ','.join(id_list)
TypeError: sequence item 9: expected str instance, NoneType found

**Comment by lokendras395** [comment]
yes when i search for "cahinsmokers" it crashes with following error:
"""
Searching for 'chainsmokers'                                                                                                                                            
Traceback (most recent call last):
  File "/usr/local/bin/mpsyt", line 11, in <module>
    sys.exit(main.main())
  File "/home/user/.local/lib/python3.5/site-packages/mps_youtube/main.py", line 141, in main
    if matchfunction(i.function, i.regex, userinput):
  File "/home/user/.local/lib/python3.5/site-packages/mps_youtube/main.py", line 64, in matchfunction
    func(*matches)
  File "/home/user/.local/lib/python3.5/site-packages/mps_youtube/commands/search.py", line 212, in search
    _search(term, query, msg, failmsg)
  File "/home/user/.local/lib/python3.5/site-packages/mps_youtube/commands/search.py", line 41, in _search
    loadmsg=loadmsg)
  File "/home/user/.local/lib/python3.5/site-packages/mps_youtube/commands/songlist.py", line 56, in paginatesongs
    songs = func[s:e]
  File "/home/user/.local/lib/python3.5/site-packages/mps_youtube/util.py", line 47, in __getitem__
    self.ilist.append(next(self.iterable))
  File "/home/user/.local/lib/python3.5/site-packages/mps_youtube/commands/search.py", line 28, in iter_songs
    for song in get_tracks_from_json(wdata2):
  File "/home/user/.local/lib/python3.5/site-packages/mps_youtube/commands/search.py", line 342, in get_tracks_from_json
    'id': ','.join([get_track_id_from_json(i) for i in items])}
TypeError: sequence item 9: expected str instance, dict found
"""

**Comment by ritiek** [comment]
@lokendras395 Are you still facing this after installing `mps-youtube` directly from the `develop` branch?

**Comment by drzacek** [comment]
After cloning git repo/develop and starting ./mpsyt it seems to work fine - at least doesn't crash on those searches I tried earlier. Any ETA when it will get to stable/master?

**Comment by lokendras395** [comment]
@ritiek  no, i installed it using pip.
pip3 install mps-youtube
if you want me to try using develop branch, i can do do that.

**Comment by clockworkpc** [comment]
Same here, Ubuntu 16.04.

Buggy if I install from this:
`pip3 install mps-youtube`

Smooth if I install from the develop branch.
```
git clone <develop branch>
cd mps-youtube
sudo ./setup.py install
```

**Comment by mhluska** [comment]
It still crashes for me on the develop branch.

**Comment by ritiek** [comment]
@mhluska is it random or happens with a specific search? If it is the latter, can you share the search term?

**Comment by mhluska** [comment]
Same as some of the other issues. Try "disturbed".

The error I get is:
```
    'id': ','.join([get_track_id_from_json(i) for i in items])}
TypeError: sequence item 9: expected str instance, dict found
```

My version:
```
mpsyt version      : 0.2.7.1
   notes           : released 6 July 2016
pafy version       : 0.5.3.1 (youtube-dl backend)
youtube-dl version : 2017.08.27.1
Python version     : 3.5.3 (default, Jan 19 2017, 14:11:04)
[GCC 6.3.0 20170124]
Processor          :
Machine type       : armv7l
Architecture       : 32bit, ELF
Platform           : Linux-4.9.41-v7+-armv7l-with-debian-9.1
sys.stdout.enc     : UTF-8
default enc        : utf-8
Config dir         : /home/pi/.config/mps-youtube
env:TERM           : screen-256color
env:SHELL          : /usr/bin/zsh
env:LANG           : en_US.UTF-8
```

Installed using: `sudo pip3 install -U git+https://github.com/mps-youtube/mps-youtube.git`

**Comment by DarkLight273** [comment]
you can use 'user marshmellow ' or '/marshmellow songname'

**Comment by ritiek** [comment]
@mhluska That is weird.

I just tried uninstalling existing `mpsyt` and re-installing it using `pip3 install mps-youtube` and I do get this error with `/disturbed` but after running `pip3 install -U git+https://github.com/mps-youtube/mps-youtube.git`, the error fixes for me.

Make sure `mpsyt` is not installed from `apt` (old version):

```
sudo apt-get purge mps-youtube
```

Check if it works after re-installing:

```
sudo pip3 install -U git+https://github.com/mps-youtube/mps-youtube.git
```

If still no luck. Could you try cloning the `develop` branch and running the script directly:

```
git clone https://github.com/mps-youtube/mps-youtube
cd mps-youtube
./mpsyt
```

**Comment by Vins1994** [comment]
[QUOTE=ritiek]I just tried uninstalling existing mpsyt and re-installing it using pip3 install mps-youtube and I do get this error with /disturbed but after running pip3 install -U git+https://github.com/mps-youtube/mps-youtube.git, the error fixes for me[/QUOTE]

Wow, thanks man, very much, this helped me out. 
I was using this program on FreeBSD and had become addicted to it since it came out first. Started crashing about a month ago, when I assumed that was due to the upgraded version of some python libraries. Usually I to avoid installing anything with any other tool but the built-in port managers, so as to make always the system keep track of what's installed and prevent broken dependencies, but this time for the sake of mps-youtube, I had reinstalled and upgraded mpsyt and youtube-dl using py36-pip...still with no luck.
I hadn't tried to clone it directly, as I thought it would have been the latest update anyway....but this worked it out!
